src/ESP_FlexyStepper.cpp

```diff
@@ -91,21 +91,21 @@ ESP_FlexyStepper::~ESP_FlexyStepper()
 }
 
 //TODO: use https://github.com/nrwiersma/ESP8266Scheduler/blob/master/examples/simple/simple.ino for ESP8266
-void ESP_FlexyStepper::startAsService(bool core)
+void ESP_FlexyStepper::startAsService(int core)
 {
  if(!core_0_wdt_was_disabled_from_flexyStepper){
 disableCore0WDT(); // we have to disable the Watchdog timer to prevent it from rebooting the ESP all the time another option would be to add a vTaskDelay but it would slow down the stepper
 core_0_wdt_was_disabled_from_flexyStepper=1;
   }
   
-  xTaskCreatePinnedToCore(
+  xTaskCreateUniversal(
```

Comment:
Unfortunately I am not a programmer and I have not read the documentation on freertos. But please take a look at the implementation of this function in the ESP32:

```
BaseType_t xTaskCreateUniversal( TaskFunction_t pxTaskCode,
                        const char * const pcName,
                        const uint32_t usStackDepth,
                        void * const pvParameters,
                        UBaseType_t uxPriority,
                        TaskHandle_t * const pxCreatedTask,
                        const BaseType_t xCoreID ){
#ifndef CONFIG_FREERTOS_UNICORE
    if(xCoreID >= 0 && xCoreID < 2) {
        return xTaskCreatePinnedToCore(pxTaskCode, pcName, usStackDepth, pvParameters, uxPriority, pxCreatedTask, xCoreID);
    } else {
#endif
    return xTaskCreate(pxTaskCode, pcName, usStackDepth, pvParameters, uxPriority, pxCreatedTask);
#ifndef CONFIG_FREERTOS_UNICORE
    }
#endif
}
```
You may have see that in the header file, I set the kernel number to "-1", which means this function will call `xTaskCreate`, also, thanks to `#ifndef CONFIG_FREERTOS_UNICORE`, this function will work correctly on single-core versions of the ESP32 chip anyway.
Coming back to your comment https://github.com/pkerspe/ESP-FlexyStepper/pull/10#discussion_r557313359
we see that when the condition `(xCoreID> = 0 && xCoreID <2)` is satisfied, `xTaskCreatePinnedToCore` will be called

Reply:
I see the benefits of this function, yet I am worried that it is not documented anywhere, this could result in lack of support of this function in the future or in general. So how did you find out about this function if you also did not see it in any documentation? How can we be sure this function is officially supported?

Reply:
just for reference definition of this function in git: 
https://github.com/espressif/arduino-esp32/blob/master/cores/esp32/esp32-hal-misc.c#L116

I learned so far, that this is an Arduino-ESP32 specific function from this comment: 
https://gitter.im/me-no-dev/ESPAsyncWebServer?at=5d97843e940b4c2fc0755933

couldn't find much more info about it, well, we can try it and hope it keeps getting maintained and not become deprecated at some point :-)

Reply:
One more thing:
I noticed that the disableCore1WDT / disableCore0WDT are also only available on the dual core models, so if we use the xTaskCreateUniversal function which will perform the unicore check, we also need to wrap the WDT disabled code portions into the `#ifndef` guard statements otherwise it will fail to run/compile on single core EPS32 imho.

https://github.com/espressif/arduino-esp32/blob/master/cores/esp32/esp32-hal-misc.c#L116:
```
#ifndef CONFIG_FREERTOS_UNICORE
void enableCore1WDT(){
    TaskHandle_t idle_1 = xTaskGetIdleTaskHandleForCPU(1);
    if(idle_1 == NULL || esp_task_wdt_add(idle_1) != ESP_OK){
        log_e("Failed to add Core 1 IDLE task to WDT");
    }
}

void disableCore1WDT(){
    TaskHandle_t idle_1 = xTaskGetIdleTaskHandleForCPU(1);
    if(idle_1 == NULL || esp_task_wdt_delete(idle_1) != ESP_OK){
        log_e("Failed to remove Core 1 IDLE task from WDT");
    }
}
#endif
````

Reply:
I was guided by this: https://github.com/espressif/arduino-esp32/blob/af119215353b26d944b80d7126bed68e060e6697/cores/esp32/esp32-hal-misc.c#L116

Reply:
https://github.com/espressif/arduino-esp32/blob/8d0e68db4f73c6243be4e9c5955ef1eb842dd95b/cores/esp32/esp32-hal-misc.c#L86
` disableCore0WDT `  -  is available on single core version.
`disableCore1WDT()` - we don't need it, at least for me it raises a warning the first time I try to use it.
So I don't see any problems